### PR TITLE
Meta: ConfigureComponents utility for CMake command generation

### PR DESCRIPTION
**Pull request by MaxWipfli**
This adds a utility program called `ConfigureComponents`. It can be used to easily configure which system components should be built. It is in essence a CMake command generator. It uses `whiptail(1)` for user interaction.

First, it allows you to specify which base configuration to use:
* Required
* Recommended
* Full
* Current (the configuration currently in use)

All of those can then be customized by manually adding or removing single components. In the end, the CMake command will be generated and shown to the user, after which it is executed. At the end, the program will run `ninja clean` and `rm -rf Root` to remove old build artifacts from components which may not be built anymore.

Due to the limitations of `whiptail(1)`, required components can still be manually unchecked, although they are marked with `[required]` and will be built anyway. A warning about this will be shown to the user if they deselect any required components.

A custom target has been added to `CMakeLists.txt`, which means that `ninja configure-components` will build and run the utility.

In the second commit, documentation has been added for the utilities in the build instructions.

This builds on top of #8066, which added support for the component configuration in the first place.

**Note:** I have no idea if this is a sensible workflow, and if I chose the correct CMake commands to specify all of this. Feedback welcome!

### CODE REVIEW

File: Meta/CMake/ConfigureComponents/main.cpp [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+        close(read_fd);
+        return -1;
+    }
+
+    int return_code = WEXITSTATUS(status);
+    if (return_code > 0) {
+        close(read_fd);
+        // posix_spawn returns 127 if it cannot exec the child, so maybe 'whiptail' is missing.
+        if (return_code == 127)
+            warnln("\e[31mError:\e[0m Could not exectue 'whiptail', maybe it isn't installed.");
```
Comment (gunnarbeutner) [inline Meta/CMake/ConfigureComponents/main.cpp:201]:
```suggestion
            warnln("\e[31mError:\e[0m Could not execute 'whiptail', maybe it isn't installed.");
```

**Review by gunnarbeutner — CHANGES_REQUESTED** [review]

File: Meta/CMake/ConfigureComponents/main.cpp [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+    if (!isatty(STDIN_FILENO)) {
+        warnln("Not a terminal!");
+        return 1;
+    }
+
+    auto current_working_directory = get_current_working_directory();
+    if (!current_working_directory.has_value())
+        return 1;
+
+    if (!current_working_directory->ends_with("Build/i686")) {
```
Comment (IdanHo) [inline Meta/CMake/ConfigureComponents/main.cpp:241]: Any chance we can make this non-i686 specific?

File: Meta/CMake/ConfigureComponents/main.cpp [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+
+static Optional<String> get_current_working_directory()
+{
+    char* cwd = getcwd(nullptr, 0);
+    if (!cwd) {
+        perror("getcwd");
+        return {};
+    }
+    String data { cwd };
+    delete cwd;
```
Comment (gunnarbeutner) [inline Meta/CMake/ConfigureComponents/main.cpp:55]:
```suggestion
    free(cwd);
```

**Review by gunnarbeutner — CHANGES_REQUESTED** [review]
Ideally we'd also have support for non-i686 architectures, but that can wait until another PR. :)

**Comment by MaxWipfli** [comment]
> Ideally we'd also have support for non-i686 architectures, but that can wait until another PR. :)

I removed the hardcoded `Build/i686`, it now just expects `Build/*`.

**Pull request merged**